repos/fdbt-site/tests/pages/api/managePurchaseMethod.test.ts

```diff
@@ -271,10 +272,40 @@ describe('managePurchaseMethod', () => {
 
         await managePurchaseMethod(req, res);
 
+        expect(updateSpy).not.toBeCalled();
         expect(insertSpy).toBeCalledWith('TEST', expected);
         expect(updateSessionAttributeSpy).toHaveBeenCalledWith(req, GS_PURCHASE_METHOD_ATTRIBUTE, undefined);
         expect(res.writeHead).toBeCalledWith(302, {
             Location: '/viewPurchaseMethods',
         });
     });
+
+    it('updates and redirects to /viewPurchaseMethods when a valid edit is performed', async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: {
+                id: 77,
+                purchaseLocations: ['OnBoard', 'Online Account'],
+                paymentMethods: ['Cash'],
+                ticketFormats: ['Paper Ticket', 'Debit/Credit card'],
+                name: 'a name',
+            },
+        });
+
+        const expected: SalesOfferPackage = {
+            id: 77,
+            purchaseLocations: ['OnBoard', 'Online Account'],
+            ticketFormats: ['Paper Ticket', 'Debit/Credit card'],
+            paymentMethods: ['Cash'],
+            name: 'a name',
+        };
+
+        await managePurchaseMethod(req, res);
+
+        expect(insertSpy).not.toBeCalled();
+        expect(updateSpy).toBeCalledWith('TEST', expected);
+        expect(updateSessionAttributeSpy).toHaveBeenCalledWith(req, GS_PURCHASE_METHOD_ATTRIBUTE, undefined);
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/viewPurchaseMethods',
+        });
+    });
```

Comment:
Should there be some tests for invalid edits causing errors?

Reply:
I was thinking that's covered by the existing tests as there isn't a different code path for edits in terms of validation - what do you think?

Reply:
When testing locally I'm not sure the validation's working as expected - if I edit a SOP and uncheck all options, I just get redirected to a blank /managePurchaseMethod page, rather than presented with errors?

Reply:
Good spot - more tests definitely required! Have push a fix